src/main/java/net/minecraftforge/items/IItemHandlerObserver.java

```diff
@@ -0,0 +1,12 @@
+package net.minecraftforge.items;
+
+import net.minecraft.item.ItemStack;
+
+import javax.annotation.Nonnull;
+
+public interface IItemHandlerObserver
+{
+    void onStackInserted(IItemHandler handler, @Nonnull ItemStack oldStack, @Nonnull ItemStack newStack, int slot);
```

Comment:
how does this handle a case where only the stack count changed?
also move slot argument to be the second one

Reply:
yes

Reply:
> how

what would old and newstack be if I insert 1 stone into an inventory and it stacks with an already existing stack of 1 stone? Wouldn't this force unnecessary copies if only the stack size changed? Because you could've just called `grow(1)` on the one in the inventory.

Reply:
yes, is the oldstack usefull to have?